### PR TITLE
Bar aggregator

### PR DESCRIPTION
**Pull request by milczarekIT**
Changes proposed in this pull request:
- Added BarAggregator and TimeSeriesAggregator to provide functionality proposed by @nimo23 in https://github.com/ta4j/ta4j/pull/420, but fixed issues (like sonar issue, NPE etc.). Provided solution in this PR covers object oriented approach.

- [X] added an entry to the unreleased section of `CHANGES.md`

### CODE REVIEW

**Comment by nimo23** [comment]
>milczarekIT I find your solution really better than the previous util-solution of https://github.com/ta4j/ta4j/pull/420. 

With your solution, one can provide different kind of aggregations not only based on Duration (DurationBarAggregator) but other things. Thank you.

File: ta4j-core/src/test/java/org/ta4j/core/aggregator/DurationBarAggregatorTest.java [hunk 1]
```diff
@@ -115,4 +115,16 @@ public void upscaledTo10DayBars() {
...
 
+    /**
+     * Tests if the bars are upscaled correctly from 1day to 10day, allowed not final bars too
+     */
+    @Test
+    public void upscaledTo10DayBarsNotOnlyFinalBars() {
+        final DurationBarAggregator barAggregator = new DurationBarAggregator(Duration.ofDays(10), false);
+        final List<Bar> bars = barAggregator.aggregate(getOneDayBars());
+
+        // must be 1 bars
```
Comment (nimo23) [inline ta4j-core/src/test/java/org/ta4j/core/aggregator/DurationBarAggregatorTest.java]: please change comment to `// must be 2 bars`

Comment (milczarekIT) [reply ta4j-core/src/test/java/org/ta4j/core/aggregator/DurationBarAggregatorTest.java]: Fixed

**Comment by team172011** [comment]
@nimo23 @milczarekIT I also like this well structured approach with good designed apis!

**Pull request merged**